Flux diagram/cte_replace_labels_to_smiles v2.ipynb

```diff
@@ -0,0 +1,495 @@
+    "\n",
+    "flux_diagram_folder_path=os.path.abspath(_)\n",
+    "\n",
+    "home = os.getenv(\"HOME\") or os.path.expanduser(\"~\")\n",
+    "rmg_pypath = os.path.join(home, 'anaconda3', 'envs', 'rmg_env', 'bin', 'python')\n",
+    "if not os.path.isfile(rmg_pypath):\n",
+    "    raise Error('Could not find RMG-env')\n",
+    "\n",
+    "######### Edit the path below ##############\n",
+    "command = [rmg_pypath, '/home/nelly/OneDrive/Linux backup/scripts/Cantera/Flux diagram/background_script.py', species_dict_path]\n",
+    "############################################\n",
+    "\n",
+    "completed_process = subprocess.run(command, shell=False, capture_output=True)\n",
+    "\n",
+    "print(\"the terminal traceback: \",completed_process)\n",
+    "\n",
+    "\n",
+    "############### Edit the Shock Tube initial parameter ################\n",
+    "reactorTemperature = 1310  # Kelvin\n",
+    "reactorPressure = 3.92*101325.0  # Pascals\n",
+    "#Edit the Species labels by their SMILES except of Ar+Ne elements\n",
+    "# Define the fuel, oxidizer and set the stoichiometry:\n",
+    "gas_X = {'CC1=CC=C(C)O1': 0.01, '[O][O]': 0.075, 'Ar': 0.915}\n",
+    "######################################################################"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "14440e06",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with open(output_path, 'r') as f:\n",
+    "    species_output_dict = yaml.load(stream=f, Loader=yaml.FullLoader)\n",
+    "\n",
+    "for label in species_output_dict.keys():\n",
+    "    if label!='Ne' and label!='Ar': # to avoid brackets in the Ne and Ar elements in cti file\n",
+    "        search_text = label\n",
+    "        replace_text = species_output_dict[label] #smiles\n",
+    "        with open(cantera_annottated_path, 'r') as file:\n",
+    "            data = file.read()\n",
+    "            data = data.replace(search_text, replace_text)\n",
+    "\n",
+    "        with open(cantera_annottated_path, 'w') as file:\n",
+    "            file.write(data)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "4955313a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def convert_pdf_2_png(smiles,pdf_path):\n",
+    "    \"\"\"\n",
+    "    Cairo2D in RMG supports PDF (higher resolution than svg/png) \n",
+    "    BUT, Graphiz DOT in Cantera doesnt support PDF\n",
+    "    SO, converting PDF image to png image results in better resolution than original png\n",
+    "    \"\"\"\n",
+    "    pdf = convert_from_path(pdf_path, 300) #creates list\n",
+    "    pdf[0].save(f'{smiles}.png') #dpi=300\n",
+    "\n",
+    "\n",
+    "def pdf_smiles_2_png(species_output_dict,flux_diagram_folder_path):\n",
+    "    \"\"\"\n",
+    "    This func goes over the pdf images in images folder, and converts each pdf image to png image\n",
+    "    \"\"\"\n",
+    "    os.chdir(flux_diagram_folder_path+'/images')\n",
```

Comment:
instead of `flux_diagram_folder_path+'/images'`, can you do `os.path.join(flux_diagram_folder_path, 'images)`?